### ISSUE TITLE
Unable to integrate wallet-selector in Next app

### ISSUE THREAD

**Issue by skiran017** [near/wallet-selector#297]
### Problem
I am trying to integrate wallet-selector in my Nextjs app but it fails with error window  not defined.

![image](https://user-images.githubusercontent.com/23178403/169741565-d6dfc7e7-65bc-4b70-b97c-205cab8ee815.png)

### What I tried so far
- tried to import NearWalletSelector dynamically 
```
const NearWalletSelector = dynamic(
  () => import('@near-wallet-selector/core/'),
  {
    ssr: false,
  }
);
```

- tried using  
```
useEffect(() => {
 if (typeof window !== "undefined") {
   const selector = await NearWalletSelector.init({config...}).then((instance) => {
        return instance.getAccounts().then(async (newAccounts) => {
          syncAccountState(localStorage.getItem('accountId'), newAccounts);
         
          window.selector = instance;
          setSelector(instance);
        });
      })
      .catch((err) => {
        console.error(err);
        alert('Failed to initialise wallet selector');
      });
},[]}
}
```

Can someone help me with resolving this issue? It would be very helpful if there can be an example in Nextjs for wallet-selector.

**Comment by kujtimprenkuSQA** [comment]
Hey @skiran017 , thanks for raising this issue!

Can you please provide a demo where you reproduce the issue you're having?

**Comment by skiran017** [comment]
https://github.com/skiran017/near-wallet-selector
I have create a demo project, the content is same as on Guest book from wallet-selector example.
Clone the repo. and when we just start the dev server we would be able to see the error as mentioned in the above image.
Thank you for responding quickly.

**Comment by kujtimprenkuSQA** [comment]
Hey @skiran017 thank you for providing this demo, you're correct in this version of our library **v3.0.1** , `window` seems to cause problems with SSR projects.

We have been working recently to Decuple Web APIs more info in this [PR](https://github.com/near/wallet-selector/pull/284) and we've already made an alpha release for the [v4.0.0-alpha.0](https://www.npmjs.com/package/@near-wallet-selector/core), in this release there are no errors/warnings about the `window` object being undefined for `@near-wallet-selector/core`.

Although that has been fixed you might face another issue with our wallet packages where Next.js would throw an error: `SyntaxError: Unexpected token 'export'` you can read more about why this happens in this [link ](https://stackoverflow.com/questions/65936222/next-js-syntaxerror-unexpected-token-export)and potentially how to get around this issue.

Note: We're working on fixing the mentioned issue in the next release.

**Comment by skiran017** [comment]
@kujtimprenkuSQA  thank you for the update. Looking forward for the next release.

**Comment by kujtimprenkuSQA** [comment]
Hey @skiran017 we have released a new version [v.4.0.0-alpha.1](https://www.npmjs.com/package/@near-wallet-selector/core) which contains fixes for Next.js.

Please keep in mind that this is a major update and the way developers interact with this project has become easier.
Docs : https://github.com/near/wallet-selector/tree/dev
Examples: https://github.com/near/wallet-selector/tree/dev/examples
Packages: https://github.com/near/wallet-selector/tree/dev/packages

We will no longer provide support for v3 since the official release of  v4.0.0 will be very soon, in the meantime you can use the alpha version. 

If you have any other questions/issues please let us know.

**Comment by skiran017** [comment]
@kujtimprenkuSQA  thank you for the links, i have been going through them and was modifying my project accordingly, found out that there is no such package `@near-wallet-selector/my-near-wallet`
![image](https://user-images.githubusercontent.com/23178403/170848857-7b064496-974e-4e2f-be1b-790538f8b926.png)

**Comment by kujtimprenkuSQA** [comment]
Hey, thanks for trying out our latest version. My Near Wallet has been merged to dev branch late on Friday but it has not been published to NPM yet.

We plan to publish it very soon, for now just avoid using it. 
We will let you know when we do so.

**Comment by kujtimprenkuSQA** [comment]
Hey @skiran017 we've offically released version 4.0.0 for all packages.

Docs/ Examples: https://github.com/near/wallet-selector
Live demo: https://near.github.io/wallet-selector/

If your above issues have been resolved feel free to close this issue.

**Comment by skiran017** [comment]
Hello @kujtimprenkuSQA, thank you for the update, i am able to integrate the package without any problem:).
One other thing i found was the images for the different wallets are not being displayed
![image](https://user-images.githubusercontent.com/23178403/171105546-e5ecd033-bed6-435a-8313-84c3d416262f.png)

even though the package has images in the assets, the browser is unable to load them, the way i am using it currently is 
```
setupNearWallet({iconUrl:'/assets/near.png'})
```
the rest of the methods i am yet to implement. Closing this issue. Much appreciated for the effort and quick response.
